catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/ProcessingDetailsImpl.java

```diff
@@ -82,6 +83,20 @@ public ProcessingDetailsImpl(SourceProcessingDetails details, String sourceId) {
     return warnings;
   }
 
+  @Override
+  public boolean equals(Object o) {
```

Comment:
@kentmorrissey were these `equals()` and `hashcode()` functions generated by IntelliJ?

Reply:
I don't think I should omit this method because the inclusion of this method allows that, where applicable, tests can verify that `exception`s and `sourceId`s match in addition to `warnings`.